service/history/handler.go

```diff
@@ -1295,6 +1295,44 @@ func (h *Handler) ReplicateRawEvents(
 	return nil
 }
 
+// ReplicateEventsV2 is called by processor to replicate history events for passive domains
+func (h *Handler) ReplicateEventsV2(
+	ctx context.Context,
+	replicateRequest *hist.ReplicateEventsV2Request,
+) (retError error) {
+
+	defer log.CapturePanic(h.GetLogger(), &retError)
+	h.startWG.Wait()
+
+	scope := metrics.HistoryReplicateEventsV2Scope
+	h.metricsClient.IncCounter(scope, metrics.CadenceRequests)
+	sw := h.metricsClient.StartTimer(scope, metrics.CadenceLatency)
+	defer sw.Stop()
+
+	domainID := replicateRequest.GetDomainUUID()
+	if domainID == "" {
+		return h.error(errDomainNotSet, scope, domainID, "")
+	}
+
+	if ok := h.rateLimiter.Allow(); !ok {
+		return h.error(errHistoryHostThrottle, scope, domainID, "")
+	}
+
+	workflowExecution := replicateRequest.WorkflowExecution
+	workflowID := workflowExecution.GetWorkflowId()
+	engine, err1 := h.controller.GetEngine(workflowID)
```

Comment:
do not really need `err1` and `err2` here.. I know this is a pattern, but should start getting rid of unnecessary patterns like this.

Reply:
this is not a pattern, this is due to people copy paste code